src/History.tsx

```diff
@@ -24,22 +26,30 @@ export class History extends React.Component<Props, {}> {
 
     render() {
         return (
-            <div className="wc-message-groups" ref={ref => this.scrollMe = ref} onScroll={ e => this.props.actions.setAutoscroll((e.target as any).scrollTop + (e.target as any).offsetHeight >= (e.target as any).scrollHeight }>
+            <div className="wc-message-groups" ref={ ref => this.scrollMe = ref } onScroll={ e => this.props.actions.setAutoscroll((e.target as any).scrollTop + (e.target as any).offsetHeight >= (e.target as any).scrollHeight) }>
                 <div className="wc-message-group">
                 { this.props.activities
                     .filter(activity => activity.type === "message" && (activity.from.id != this.props.userId || !activity.id))
-                    .map((activity:Message) => 
-                        <div className={ 'wc-message wc-message-from-' + (activity.from.id === 'user' ? 'me' : 'bot') }>
-                            <div className="wc-message-content">
-                                <svg className="wc-message-callout">
-                                    <path className="point-left" d="m0,0 h12 v10 z" />
-                                    <path className="point-right" d="m0,10 v-10 h12 z" />
-                                </svg>
-                                <HistoryMessage activity={ activity } actions={ this.props.actions }/>
+                    .map((activity:Message) => {
+                        let clickable = this.props.debuggerVisible;
+                        let selected = activity === this.props.selectedActivity;
+                        let contentClassName = 'wc-message-content' + (clickable ? ' clickable' : '') + (selected ? ' selected' : '');
```

Comment:
For different reasons, these locals don't really buy anything. Let's get rid of them.